server/src/service.rs

```diff
@@ -76,6 +80,14 @@ impl Service {
             .await
             .insert(path.clone(), session.clone());
 
+        if let Some(sender) = header.sender() {
+            tracing::info!("Client {} connected.", sender);
+            self.clients
+                .lock()
+                .await
+                .insert(sender.as_str().to_owned(), path.clone());
```

Comment:
Why do you convert the sender type to a string?

Reply:
Also why doesn't this keep a ref to the session instead?

Reply:
I feel like storing both the session twice is too much, maybe we should combine the path / sender into a key for finding the corresponding session

Reply:
Yeah, I agree. Session now stores the client_name and introduced session_from_client_name

Reply:
> Why do you convert the sender type to a string?

UniqueName requires named lifetime parameter :)

Reply:
There is an OwnedUniqueName. Please don't mark threads as resolved when they are not.